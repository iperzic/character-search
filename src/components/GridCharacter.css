.GridCharacter {
    position: relative;
    height: 220px;
    width: 100%;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 4px 20px 0 rgba(0, 0, 0, 0.19);
}

.GridCharacter__image {
    height: 220px;
    width: 100%;
    object-fit: cover;
}

.GridCharacter__name {
    position: absolute;
    bottom: 0;
    width: calc(100% - 20px);
    padding: 10px;
    background-color: rgba(255, 255, 255, 0.4);
    margin-bottom: 0;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.GridCharacter__bookmark {
    position: absolute;
    top: 0;
    right: 0;
    padding: 5px;
    cursor: pointer;
    font-size: 1.2em;
    color: gold;
    text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
}

.GridCharacter__bookmark:hover {
    opacity: 0.7;
}

.bookmark--not-selected:after {
    content: '☆';
}

.bookmark--selected:after {
    content: '★';
}
