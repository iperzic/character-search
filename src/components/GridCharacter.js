import React from 'react';
import PropTypes from 'prop-types';

import './GridCharacter.css';

const GridCharacter = ({ name, image, isBookmark, onBookmarkClick }) => (
  <div className="GridCharacter">
    <img className="GridCharacter__image" src={image} alt={name} />
    <h2 className="GridCharacter__name" title={name}>{name}</h2>
    <span
      className={isBookmark ?
        'GridCharacter__bookmark bookmark--selected' :
        'GridCharacter__bookmark bookmark--not-selected'}
      onClick={onBookmarkClick}
    />
  </div>
);

GridCharacter.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  isBookmark: PropTypes.bool.isRequired,
  onBookmarkClick: PropTypes.func.isRequired,
};

export default GridCharacter;
