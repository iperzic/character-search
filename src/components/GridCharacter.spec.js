import React from 'react';
import { shallow } from 'enzyme';

import GridCharacter from './GridCharacter';

describe('GridCharacter', () => {
  let component;
  let onBookmarkSpy;
  let name;
  let image;
  let isBookmark;

  beforeEach(() => {
    onBookmarkSpy = jest.fn();
    name = 'Deadpool';
    image = 'http://some-deadpool-image';
    isBookmark = true;
    component = shallow(
      <GridCharacter
        name={name}
        image={image}
        isBookmark={isBookmark}
        onBookmarkClick={onBookmarkSpy}
      />);
  });

  describe('when the character is bookmarked', () => {
    it('should display that the character is bookmarked', () => {
      expect(component.getElement()).toMatchSnapshot();
    });
  });

  describe('when the character is not bookmarked', () => {
    it('should display that the character is not bookmarked', () => {
      component.setProps({isBookmark: false});

      expect(component.getElement()).toMatchSnapshot();
    });
  });

  describe('when bookmark icon is clicked', () => {
    it('should call the onClick prop if present', () => {
      component.find('.GridCharacter__bookmark').simulate('click');

      expect(onBookmarkSpy).toHaveBeenCalledTimes(1);
    });
  });
});
