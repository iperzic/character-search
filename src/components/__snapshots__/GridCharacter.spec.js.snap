// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridCharacter when the character is bookmarked should display that the character is bookmarked 1`] = `
<div
  className="GridCharacter"
>
  <img
    alt="Deadpool"
    className="GridCharacter__image"
    src="http://some-deadpool-image"
  />
  <h2
    className="GridCharacter__name"
    title="Deadpool"
  >
    Deadpool
  </h2>
  <span
    className="GridCharacter__bookmark bookmark--selected"
    onClick={[Function]}
  />
</div>
`;

exports[`GridCharacter when the character is not bookmarked should display that the character is not bookmarked 1`] = `
<div
  className="GridCharacter"
>
  <img
    alt="Deadpool"
    className="GridCharacter__image"
    src="http://some-deadpool-image"
  />
  <h2
    className="GridCharacter__name"
    title="Deadpool"
  >
    Deadpool
  </h2>
  <span
    className="GridCharacter__bookmark bookmark--not-selected"
    onClick={[Function]}
  />
</div>
`;
